phpunit/functional/Glpi/Form/ConditionalVisibility/EngineTest.php

```diff
@@ -259,8 +259,159 @@ public static function conditionsOnComments(): iterable
+        $form->addComment("Comment 2");
+        $form->setSectionVisibility(
+            "Test section 1",
+            VisibilityStrategy::VISIBLE_IF,
+            [
+                [
+                    'logic_operator' => LogicOperator::AND,
+                    'item_name'      => "Question 1",
+                    'item_type'      => Type::QUESTION,
+                    'value_operator' => ValueOperator::EQUALS,
+                    'value'          => "answer for question 1",
+                ]
+            ]
+        );
+        $form->setSectionVisibility(
+            "Test section 2",
+            VisibilityStrategy::HIDDEN_IF,
+            [
+                [
+                    'logic_operator' => LogicOperator::AND,
+                    'item_name'      => "Question 2",
+                    'item_type'      => Type::QUESTION,
+                    'value_operator' => ValueOperator::EQUALS,
+                    'value'          => "answer for question 2",
+                ]
+            ]
+        );
+
+        yield [
+            'form' => $form,
+            'input' => [
+                'answers' => [
+                    'Question 1' => "",
+                    'Question 2' => "",
+                ],
+            ],
+            'expected_output' => [
+                'sections' => [
+                    'Test section 1' => false,
+                    'Test section 2' => true,
+                ],
+            ],
+        ];
+        yield [
+            'form' => $form,
+            'input' => [
+                'answers' => [
+                    'Question 1' => "answer for question 1",
+                    'Question 2' => "",
+                ],
+            ],
+            'expected_output' => [
+                'sections' => [
+                    'Test section 1' => true,
+                    'Test section 2' => true,
+                ],
+            ],
+        ];
+        yield [
+            'form' => $form,
+            'input' => [
+                'answers' => [
+                    'Question 1' => "",
+                    'Question 2' => "answer for question 2",
+                ],
+            ],
+            'expected_output' => [
+                'sections' => [
+                    'Test section 1' => false,
+                    'Test section 2' => false,
+                ],
+            ],
+        ];
+        yield [
+            'form' => $form,
+            'input' => [
+                'answers' => [
+                    'Question 1' => "answer for question 1",
+                    'Question 2' => "answer for question 2",
+                ],
+            ],
+            'expected_output' => [
+                'sections' => [
+                    'Test section 1' => true,
+                    'Test section 2' => false,
+                ],
+            ],
+        ];
+    }
+
+    public static function firstSectionShouldAlwaysBeVisible(): iterable
+    {
+        $form = new FormBuilder();
```

Comment:
I had to read the FormBuilder class to make sure that this section is automatically created.

```suggestion
        $form = new FormBuilder();
        $form->addSection("First section");
```